### PR TITLE
Patch OpenSSL for ARM64/M1 Macs

### LINKED ISSUE #2: Mac M1 chntpw not building (Openssl 1.0 requirement not satisfied)
Hello.

I have a mac M1 mba and I am trying to install sidneys/homebrew/chntpw

i have tried via rosetta and also via regular arm homebrew. unfortunately there is an issue building openssl 1.0 for this cpu,

there seems to be some instructions to build it manually on openssl github repo, any chance this can be incorporated into your script?

### PR DESCRIPTION
**Pull request by itsmattkc**
Fixes #2

Uses [this patch](https://gist.githubusercontent.com/felixbuenemann/5f4dcb30ebb3b86e1302e2ec305bac89/raw/b339a33ff072c9747df21e2558c36634dd62c195/openssl-1.0.2u-darwin-arm64.patch) to add ARM64 support to OpenSSL's Configure script and the Ruby constant `#{Hardware::CPU.arch}` to correctly select the current architecture (which is also used on [homebrew-core's OpenSSL formula](https://github.com/Homebrew/homebrew-core/blob/master/Formula/openssl%401.1.rb#L92)).

### CODE REVIEW

**Comment by ItzSwirlz** [comment]
Matt, just some friendly advice (not trying to be an annoying creep or anything, it also may make sense that @sidneys should know), you may want to run RuboCop or some other Ruby linter, linters exist for basically anything so if you want to run it feel free, you can relax on the " and use ' instead (trying to say this respectfully although it'll probably sound mean)

**Comment by itsmattkc** [comment]
Hmm? Have I done something wrong?

**Comment by ItzSwirlz** [comment]
No, the code will still work. It's fine, linting is just a way of code cleanup. The PR so far is acceptable to merge. Just can't find a way to privately message you two or whatnot. If you want to just ignore what I said and that's fine.

**Comment by itsmattkc** [comment]
Are you saying the double quotes are incorrect? That's how patch code is written on every brew script: https://github.com/Homebrew/homebrew-core/blob/master/Formula/jemalloc.rb#L27

**Comment by ItzSwirlz** [comment]
It's correct, it's just code style that can be improved. I didn't mean to confuse you, just forget what I said :P

**Comment by themacmarketer** [comment]
Can someone please advise me how I can apply this patch manually ?

Thanks !

**Comment by pixluser** [comment]
> Can someone please advise me how I can apply this patch manually ?
> 
> Thanks !

pico /opt/homebrew/Library/Taps/sidneys/homebrew-homebrew/Formula/openssl@1.0.rb

1>
after the line 
 'version_scheme 1'

and between the line  
'bottle do'


add that :

 patch do
    url "https://gist.githubusercontent.com/felixbuenemann/5f4dcb30ebb3b86e1302e2ec305bac89/raw/b339a33ff072c9747df21e2558c36634dd62c195/openssl-1.0.2u-darwin-arm64.patch"
    sha256 "4ad22bcfc85171a25f035b6fc47c7140752b9ed7467bb56081c76a0a3ebf1b9f"
  end

2> change this line 

    arch_args = %w[darwin64-x86_64-cc enable-ec_nistp_64_gcc_128]

by
    arch_args = %W[darwin64-#{Hardware::CPU.arch}-cc enable-ec_nistp_64_gcc_128]



and redo your brew install.

it works on my m1.

bye !

**Comment by pixluser** [comment]
> Can someone please advise me how I can apply this patch manually ?
> 
> Thanks !

All the infos came from that url : https://github.com/sidneys/homebrew-homebrew/pull/4/commits/271a5783cd83ac160d936ead9a14d4b305d0f216

bye !

**Comment by Kakise** [comment]
This PR works amazingly on my MBP M1 on macOS 12, is there a reason why it isn't merged yet ? @sidneys ?

**Comment by shaz01** [comment]
I've done pixluser's instructions, and it showed me this error:

`Last 15 lines from /Users/marple/Library/Logs/Homebrew/chntpw/01.make:
2021-06-30 13:23:56 +0300

make

gcc -c -DUSEOPENSSL -g -I. -I/usr/local/opt/openssl/include -Wl chntpw.c
gcc -c -DUSEOPENSSL -g -I. -I/usr/local/opt/openssl/include -Wl ntreg.c
gcc -c -DUSEOPENSSL -g -I. -I/usr/local/opt/openssl/include -Wl edlib.c
gcc -c -DUSEOPENSSL -g -I. -I/usr/local/opt/openssl/include -Wl cpnt.c
gcc -c -DUSEOPENSSL -g -I. -I/usr/local/opt/openssl/include -Wl reged.c
gcc -DUSEOPENSSL -g -I. -I/usr/local/opt/openssl/include -Wl -o cpnt cpnt.o -L/usr/local/opt/openssl/lib /usr/local/opt/openssl/lib/libcrypto.a
clang: error: no such file or directory: '/usr/local/opt/openssl/lib/libcrypto.a'
`
I know the problem is that it tries to use the Intel Homebrew's folder (usr/local/opt) but m1 Homebrew's folder is opt/homebrew. Is this related to the patch? (And how can I prevent this from happening?)

**Comment by SebDanielsson** [comment]
Any reason why OpenSSL 1.1.1 isn't used? Isn't 1.0.2 unmaintained?

File: Formula/openssl@1.0.rb [hunk 1]
```diff
@@ -6,6 +6,11 @@ class OpensslAT10 < Formula
   sha256 "ecd0c6ffb493dd06707d38b14bb4d8c2288bb7033735606569d8f90f89669d16"
   version_scheme 1
 
+  patch do
+    url "https://gist.githubusercontent.com/felixbuenemann/5f4dcb30ebb3b86e1302e2ec305bac89/raw/b339a33ff072c9747df21e2558c36634dd62c195/openssl-1.0.2u-darwin-arm64.patch"
+    sha256 "4ad22bcfc85171a25f035b6fc47c7140752b9ed7467bb56081c76a0a3ebf1b9f"
+  end
+
```
[the remark this answers is not in the document]
Comment (itsmattkc) [reply Formula/openssl@1.0.rb]: Done

**Comment by ChrisDev8** [comment]
This works flawlessly on MacOS M1 if combined with Pull Request #5. This fix is greatly going to help with uup dump!!! :) Thank you!!!

**Comment by omerien** [comment]
Amazing to know that one of my favourite YouTubers help me installing brew tools xD

**Pull request merged**